src/metabase/util/malli/registry.cljc

```diff
@@ -1,13 +1,45 @@
 (ns metabase.util.malli.registry
   (:refer-clojure :exclude [def])
   (:require
+   [clojure.test.check.generators :as gen]
    [malli.core :as mc]
+   [malli.generator :as mg]
+   [malli.impl.regex :as re]
    [malli.registry :as mr]
    [malli.util :as mut])
   #?(:cljs (:require-macros [metabase.util.malli.registry])))
 
+;;; Implementation of :vcatn schema: stolen from malli sources,
+;;; would be nice to find a more composable way.
+(defn- schemas []
+  (let [-re-min-max @#'mc/-re-min-max]
+    {:vcatn (mc/-sequence-entry-schema
```

Comment:
Maybe it should be `catnv`, to rhyme with `mapv`.

Reply:
`:tuple` is vectors-only `:cat`. I don't think there's a vectors-only `:catn`, but if there were to be one I'd call it `:tuplen`.

Reply:
@bshepherdson Isn't `:tuple` for fixed length vectors?

Reply:
I think this is unnecessary anyway - it can be replaced with `[:catn {:gen/fmap vec} ...]` everywhere.

Reply:
Cool, I could only find `:gen/gen`. I'll swicth.

Reply:
Using `:gen/fmap` results in
```
   Cannot generate values due to infinitely expanding schema: [:catn #:gen{:fmap
   #function[clojure.core/vec]} [:clause [:= :not]] [:options
   :metabase.lib.schema.common/options] [:args [:schema [:ref
   :metabase.lib.schema.filter/filter]]]]
```
which is something I got earlier too when I was trying to replace just the generator.

Reply:
As for `:tuple`: I just didn't want to use different types even if I didn't have variable length args everywhere.

Reply:
We decided to go with this for now and clean up in a second step.